lib/elixir/lib/agent.ex

```diff
@@ -18,37 +18,27 @@ defmodule Agent do
       defmodule Mix.TasksServer do
         use Agent
 
-        def start_link(_) do
-          Agent.start_link(fn -> MapSet.new() end, name: __MODULE__)
+        def start_link(_opts) do
+          Agent.start_link(fn -> %{} end, name: __MODULE__)
         end
 
-        @doc "Checks if the task has already executed"
-        def executed?(task, project) do
-          item = {task, project}
-
-          Agent.get(__MODULE__, fn set ->
-            item in set
-          end)
+        def clear() do
+          update(fn _ -> %{} end)
         end
 
-        @doc "Marks a task as executed"
-        def put_task(task, project) do
-          item = {task, project}
-          Agent.update(__MODULE__, &MapSet.put(&1, item))
+        def put(tuple) do
+          update(&Map.put(&1, tuple, true))
         end
 
-        @doc "Resets the executed tasks and returns the previous list of tasks"
-        def take_all() do
-          Agent.get_and_update(__MODULE__, fn set ->
-            {Enum.into(set, []), MapSet.new()}
-          end)
+        defp update(fun) do
+          Agent.update(__MODULE__, fun, 30000)
         end
       end
 
-  Agents provide a segregation between the client and server APIs (similar
-  to GenServers). In particular, the anonymous functions given to the `Agent`
-  are executed inside the agent (the server). This distinction is important
-  because you may want to avoid expensive operations inside the agent,
+  Agents provide a segregation between the client and server APIs (similar to
+  GenServers). In particular, the functions passed as arguments to the `Agent`
```

Comment:
I find “to the `Agent` function calls” a bit confusing. What do you think about?

> In particular, the functions passed as arguments to the function calls of `Agent` are invoked inside the agent (the server).

Reply:
Sounds good!

Reply:
On a second read, "the function calls of Agent" does not read well to me, because `Agent` has functions, not function calls.

In the original sentence we refer to "`Agent` function calls", meaning calls to `Agent` functions. 

WDYT?

Reply:
“the calls to `Agent` functions”?